### ISSUE TITLE
issues on dst

### ISSUE THREAD

**Issue by smartyfh** [salesforce/simpletod#5]
Hi, 

when evaluating the JGA for DST, did you remove both the none slot and dontcare slot?

When I ran the dialogue_generation.py, it seems that the generated belief states are always empty in the MODEL_OUTPUT file. so could you please provide more details about how the model is trained for DST? 

Thanks!

**Comment by ehosseiniasl** [comment]
Hi,

We are only removing none. it is fixed now.
On dialogue_generation.py, this seems a parsing issue. We will fix this

**Comment by fasterbuild** [comment]
@smartyfh empty belief issue: add text=text.strip() before tokenizer.encode(text), or there is always an space at the end. However I didn't get the joint acc in the paper, will you?

**Comment by smartyfh** [comment]
> @smartyfh empty belief issue: in generate_dialogue.py file, add text=text.strip() before tokenizer.encode(text), or there is always an space at the end. However I didn't get the joint acc in the paper, will you?

@fasterbuild Have you checked all the checkpoints or just one checkpoint? If ignoring both none and dontcare slots, the results should be reproducible. However, if keeping the dontcare slots, the acc would go down several points. But this needs the author to confirm.

**Comment by gungui98** [comment]
@smartyfh I am struggling in reproducing the result, could you please share the hyper-params you use for the training?

**Comment by ShaneTian** [comment]
> > @smartyfh empty belief issue: in generate_dialogue.py file, add text=text.strip() before tokenizer.encode(text), or there is always an space at the end. However I didn't get the joint acc in the paper, will you?
> 
> @fasterbuild Have you checked all the checkpoints or just one checkpoint? If ignoring both none and dontcare slots, the results should be reproducible. However, if keeping the dontcare slots, the acc would go down several points. But this needs the author to confirm.

If keeping the `dontcare` slots, what is the JGA you get?

**Comment by smartyfh** [comment]
> > > @smartyfh empty belief issue: in generate_dialogue.py file, add text=text.strip() before tokenizer.encode(text), or there is always an space at the end. However I didn't get the joint acc in the paper, will you?
> > 
> > 
> > @fasterbuild Have you checked all the checkpoints or just one checkpoint? If ignoring both none and dontcare slots, the results should be reproducible. However, if keeping the dontcare slots, the acc would go down several points. But this needs the author to confirm.
> 
> If keeping the `dontcare` slots, what is the JGA you get?

about 52

**Comment by HuangLK** [comment]
> > > @smartyfh empty belief issue: in generate_dialogue.py file, add text=text.strip() before tokenizer.encode(text), or there is always an space at the end. However I didn't get the joint acc in the paper, will you?
> > 
> > 
> > @fasterbuild Have you checked all the checkpoints or just one checkpoint? If ignoring both none and dontcare slots, the results should be reproducible. However, if keeping the dontcare slots, the acc would go down several points. But this needs the author to confirm.
> 
> If keeping the `dontcare` slots, what is the JGA you get?

The JGA is 50.32% if keeping the `dontcare`. After ignoring both `none` and `dontcare`, I can achieve 55.45% JGA.

**Comment by libing125** [comment]
I got 50.46 joint accuracy, keeping dontcare and doing default cleaning.

**Comment by searchforpassion** [comment]
> @smartyfh empty belief issue: in generate_dialogue.py file, add text=text.strip() before tokenizer.encode(text), or there is always an space at the end. However I didn't get the joint acc in the paper, will you?

@fasterbuild I tried text=text.strip() before tokenizer.encode(text), it doesn't work.
